src/OpenSage.Game/Terrain/TerrainLogic.cs

```diff
@@ -1,7 +1,16 @@
-﻿namespace OpenSage.Terrain
+﻿using OpenSage.Data.Map;
+
+namespace OpenSage.Terrain
 {
     public sealed class TerrainLogic : IPersistableObject
     {
+        public HeightMap HeightMap { get; private set; }
```

Comment:
It feels weird to me that both TerrainLogic and Terrain are now holding separate references to a HeightMap - is this intended?

Reply:
I think it’s the same object instance shared by both? I needed access to it from PhysicsBehavior, but I can’t create a Terrain instance since that has a load of graphics stuff. And I also didn’t want to remove Terrain.HeightMap because it had a lot of references. At some point it should be refactored though.

Reply:
Sorry by separate references I meant separate references to the same object - just feels odd Terrain can't call into TerrainLogic or vv

Reply:
It's not a big deal though - I think the testability here is the big win!

Reply:
I guess I could pass `TerrainLogic` into `Terrain` - but ideally we should move things in `Terrain` into either `TerrainLogic` or `TerrainVisual`.